src/test/java/com/palantir/docker/compose/DockerComposeRuleShould.java

```diff
@@ -265,6 +277,32 @@ public void when_remove_conflicting_containers_on_startup_is_set_to_false_before
         rule.before();
     }
 
+    @Test
+    public void union_cluster_waits_from_builder_instead_of_overwriting() {
+        Collection<ClusterWait> originalAssignment = ImmutableList.of(
+                mock(ClusterWait.class),
+                mock(ClusterWait.class));
+        Collection<ClusterWait> secondAssignment = ImmutableList.of(mock(ClusterWait.class));
+
+        DockerComposeRule twoAssignments = defaultBuilder()
+                .clusterWaits(originalAssignment)
+                .clusterWaits(secondAssignment)
+                .build();
+
+        assertThat(twoAssignments.clusterWaits(), hasSize(3));
```

Comment:
Although in theory 3 is sufficient we should probably test what the test says we are xD